File: proj_1.py
#!/usr/bin/env python

# IRA
# 100 - CSCO
# 210 - GE

#  50 - AAPl
# 110 - BP
# 100 - CBI
#  50 - DE
# 100 - M
# 100 - MSFT
# 100 - PG
#  50 - CRM

import sys
import oauth2 as oauth
import time

import sqlite3 as lite
from database_function import *
from tk_functions import *
from spreadsheet_functions import *

commision_flat_rate = 4.95
commision_per_contract = 0.65

outer_repeat_time = 120.0
inner_repeat_time = 20.0

stock_list = [ 'AAPL',
               # 'BP',
               # 'CBI',
               # 'DE',
               # 'M',
               # 'MSFT',
               # 'PG',
               'CRM' ]

def main():
    tk_func = TK_functions()

    con = lite.connect('trade_test.db')
    with con:
        cur = con.cursor()
        setup_tables(cur, stock_list )

    # initialize workbook
    # trade_workbook = pyxl.Workbook()
    # trade_sheet = trade_workbook.active
    # trade_sheet.title = "Trade List"
    # setup_trade_columns( trade_sheet )
    # for stock in stock_list:
    #     stock_sheet = trade_workbook.create_sheet(stock)
    #     setup_columns(stock_sheet)

    # trade_workbook.save("data_dump.xls")

    # start_time=time.time()
    # while True:

    tk_func.refresh_account_data()

    for stock in stock_list:
        tk_func.get_quote(stock)
        tk_func.get_search(stock)

        # fill_row ( trade_workbook[stock],
        #         tk_func.search.search_quote,
        #         tk_func.quote.bid,
        #         tk_func.quote.ask )

        add_quote(cur,
                  stock,
                  tk_func.search.search_quote,
                  tk_func.quote.bid,
                  tk_func.quote.ask)

        find_profitable_trades( cur,
                                stock,
                                tk_func.search.search_quote,
                                tk_func.quote.bid,
                                tk_func.quote.ask )


        sql_text = "SELECT * FROM " + stock
        sql_text += " LIMIT 10"
        cur.execute(sql_text)
        rows = cur.fetchall()

        # for row in rows:
            # print row

        con.commit()
        print( 'done {} - {}'.format(stock, time.time()))
            # time.sleep(inner_repeat_time - ((time.time() - start_time) % inner_repeat_time))

        # trade_workbook.save("data_dump.xls")
        # time.sleep(outer_repeat_time - ((time.time() - start_time) % outer_repeat_time))

    # sql_text = "SELECT * FROM trades"
    # cur.execute(sql_text)
    # rows = cur.fetchall()

    # for row in rows:
    #     print row
    # trade_workbook.save("data_dump.xls")

if __name__ == '__main__':
    main()

File: proj_rb/proj_1.rb
#!/usr/lib/ruby

# IRA
# 100 - CSCO
# 210 - GE

#  50 - AAPL
# 110 - BP
# 100 - CBI
#  50 - DE
# 100 - M
# 100 - MSFT
# 100 - PG
#  50 - CRM

require_relative 'timer'
require_relative 'tk_functions'
require_relative 'spreadsheet_functions'
require 'rubygems'
require 'spreadsheet'

Spreadsheet.client_encoding = 'UTF-8'
tk_data = TK_functions.new
data_store = Spreadsheet::Workbook.new

$commision_flat_rate=4.95
$commision_per_contract=0.65
stock_array = [ "CSCO"]
                # "GE",
                # "AAPL",
                # "BP",
                # "CBI",
                # "DE",
                # "M",
                # "MSFT",
                # "PG",
                # "CRM",
                # "SPY"]

# initialize each worksheet
data_store.create_worksheet :name => "Trade List"
setup_trade_colums(data_store.worksheet("Trade List"))
for stock in stock_array
    data_store.create_worksheet :name => stock.to_s
    setup_columns(data_store.worksheet(stock.to_s))
end

tk_data.refresh_account_data()

# Timer.every 60, 60 do
# 2.times do

    for stock in stock_array
        tk_data.get_search(stock.to_s)
        tk_data.get_quote(stock.to_s)

        fill_row( data_store.worksheet(stock.to_s),
                    tk_data.search[(stock.to_s)],
                    tk_data.quote.bid,
                    tk_data.quote.ask)

        find_profitable_trades(data_store.worksheet("Trade List"),
                    data_store.worksheet(stock.to_s),
                    tk_data.search[(stock.to_s)],
                    tk_data.quote.bid,
                    tk_data.quote.ask)

        puts "done #{stock}"
    end

    data_store.write "data_dump.xls"

    puts "done loop"
# end

tk_data = nil
data_store = nil
# data_store.write "data_dump.xls"

puts "done everything"
